OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -66,6 +79,38 @@ static class DMOpcodeHandlers {
             return null;
         }
 
+        public static ProcStatus? CreateTypeEnumerator(DMProcState state)
+        {
+            if (!state.Pop().TryGetValueAsPath(out var type))
+            {
+                throw new Exception($"Cannot create a type enumerator for a non-path");
+            }
+
+            DreamList list;
+
+            if (type == DreamPath.Client)
+            {
+                list = DreamList.Create(state.DreamManager.Clients);
+            }
+            else if (state.DreamManager.ObjectTree.GetObjectDefinition(type).IsSubtypeOf(DreamPath.Atom))
+            {
+                list = state.DreamManager.WorldContentsList;
+            }
+            else if (state.DreamManager.ObjectTree.GetObjectDefinition(type).IsSubtypeOf(DreamPath.Datum))
+            {
+                list = DreamList.Create(state.DreamManager.Datums);
+            }
```

Comment:
A good optimization here would be to create a new enumerator that can enumerate a list of DreamObjects and return them as DreamValues. Using that here would remove the need to create a new DreamList.